File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public: // bk b
    bool areAlmostEqual(string s1, string s2) {
        int n=s1.length();
        int cnt=0;
        string f1="";
        string f2="";
        for(int i=0;i<n;i++){
            if(s1[i]!=s2[i]){
                cnt++;
                
                    f1+=s1[i];
                    f2+=s2[i];
                
                
                    f1+=s1[i];
                    f2+=s2[i];
                
            }
        }
        if(cnt==2){
            sort(f1.begin(),f1.end());
            sort(f2.begin(),f2.end());
            return f1==f2;
        }
        return cnt==0;
    }
};

File: 1915-check-if-one-string-swap-can-make-strings-equal/check-if-one-string-swap-can-make-strings-equal.py
class Solution(object):
    def areAlmostEqual(self, s1, s2):
        s=""
        t=""
        n=len(s1)
        for i in range(n):
            if s1[i]!=s2[i]:
                s+=s1[i]
                t+=s2[i]
            
        if len(s)==0:
            return True
        if len(s)!=2:
            return False
        
        print(s)
        rev=''.join(reversed(t))
        return True if rev==s else False
        
        